source/extensions/filters/udp/dns_filter/dns_parser.cc

```diff
@@ -97,9 +97,39 @@ bool DnsAnswerRecord::serialize(Buffer::OwnedImpl& output) {
       output.writeBEInt<uint16_t>(4);
       output.writeLEInt<uint32_t>(ip_address->ipv4()->address());
     }
-    return true;
   }
-  return false;
+  return (output.length() > 0);
+}
+
+bool DnsSrvRecord::serialize(Buffer::OwnedImpl& output) {
+  if (!targets_.empty()) {
+    // The Service Record being serialized should have only one target
```

Comment:
can this be asserted?

Reply:
It's possible that we don't have any answers to serialize which results in an NXDOMAIN response to the client.